### PR TITLE
Add documentation for shared pointer class

### PR DESCRIPTION
**Pull request by donatieng**
This is the associated documentation PR for https://github.com/ARMmbed/mbed-os/pull/7815.

### CODE REVIEW

**Comment by AnotherButler** [comment]
@jen3andruska Could you please copy edit this file when you get a chance?

**Comment by jen3andruska** [comment]
@AnotherButler  No problem!

**Comment by jen3andruska** [comment]
@AnotherButler  Finished copy-editing and made revisions.

**Comment by AnotherButler** [comment]
@jen3andruska Thanks for the CE. It looks great 👍

File: docs/reference/api/platform/SharedPtr.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+A shared pointer is a "smart" pointer that retains ownership of an object by using reference counting accross all smart pointers referencing that object.
+
+It is similar to the `std::shared_ptr` class introduced in C++11. However, this is not a compatible implementation, as there are no weak pointers, no `make_shared`, no custom deleters, and so on.
+
+Usage: `SharedPtr<Class> ptr(new Class())`
+
+When `ptr` is passed around by a value, the copy constructor and destructor manage the reference count of the raw pointer. If the counter reaches zero, `delete` is called on the raw pointer.
+
+To avoid loops, use "weak" references by calling the original pointer directly with `ptr.get()`.
```
Comment (pan-) [inline docs/reference/api/platform/SharedPtr.md]: I would remove that sentence as it is misleading and can lead to access of a dangling pointer as it is not possible to know if a pointer acquired previously with `ptr.get()` is valid or not.

**Comment by jen3andruska** [comment]
@AnotherButler  No Problem 👍

**Comment by donatieng** [comment]
Thanks @jen3andruska @pan- @AnotherButler

File: docs/reference/api/platform/SharedPtr.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+## Shared Pointer
+
+A shared pointer is a "smart" pointer that retains ownership of an object by using reference counting accross all smart pointers referencing that object.
```
Comment (cmonr) [inline docs/reference/api/platform/SharedPtr.md]: Could it be clarified what `that` is suppose to mean in this sentence?

File: docs/reference/api/platform/SharedPtr.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+It is similar to the `std::shared_ptr` class introduced in C++11. However, this is not a compatible implementation, as there are no weak pointers, no `make_shared`, no custom deleters, and so on.
+
+Usage: `SharedPtr<Class> ptr(new Class())`
+
+When `ptr` is passed around by a value, the copy constructor and destructor manage the reference count of the raw pointer. If the counter reaches zero, `delete` is called on the raw pointer.
```
Comment (cmonr) [inline docs/reference/api/platform/SharedPtr.md]: s/manage/manages

**Comment by cmonr** [comment]
@pan-  Mind re-reviewing? Looks like GitHub didn't catch the update (for some reason)

**Comment by AnotherButler** [comment]
FYI @MarceloSalazar @ashok-rao

**Comment by AnotherButler** [comment]
ping @pan-

**Pull request merged**